test/validations/relaxng/run.py

```diff
@@ -20,14 +21,26 @@
 relaxng_doc = etree.parse('test/validations/relaxng/schema.xml')
 relaxng = etree.RelaxNG(relaxng_doc)
 
+ruleset_whitelist = None
+
+with open('utils/ruleset-whitelist.csv', 'r') as ruleset_whitelist_file:
+    ruleset_whitelist_reader = csv.reader(ruleset_whitelist_file)
+    ruleset_whitelist = list(ruleset_whitelist_reader)
+
 exit_code = 0
 
 print("Validation of rulesets against relaxng schema.xml begins...")
 
 for filename in sorted(files):
+    basename = os.path.basename(filename)
+    skip_relaxng = False
+    for entry in ruleset_whitelist:
+        if entry[2] == '1' and entry[3] == basename:
+            skip_relaxng = True
+            break
     tree = etree.parse(filename)
```

Comment:
You can also check `skip_relaxng` and `continue` here to avoid unnecessary parsing `filename` into a XML tree. This will futher improve the performance.